cli/scorecard/violations.go

```diff
@@ -126,21 +129,21 @@ func getViolations(inventory *InventoryConfig, config *ScoringConfig) (*validato
 
 // converts raw JSON into Asset proto
 func getAssetFromJSON(input []byte) (*validator.Asset, error) {
-	var asset map[string]interface{}
-	err := json.Unmarshal(input, &asset)
-	if err != nil {
-		return nil, err
-	}
 	pbAsset := &validator.Asset{}
-	err = protoViaJSON(asset, pbAsset)
-	if err != nil {
-		return nil, errors.Wrapf(err, "converting asset %s to proto", asset["name"])
+	umar := &jsonpb.Unmarshaler{AllowUnknownFields: true}
+	if err := umar.Unmarshal(strings.NewReader(string(input)), pbAsset); err != nil {
+		err2 := json.Unmarshal(input, &pbAsset)
+		if err2 != nil {
+			return nil, errors.Wrap(err2, "unmarshaling to proto")
+		}
+		Log.Debug("getAssetFromJSON", "Unmarshal", "Unable to unmarshal with jsonpb but able to unmarshal with json")
```

Comment:
Handling data conversion like this because the original approach using jsonpb does not handle [update_time field](https://github.com/googleapis/googleapis/blob/master/google/cloud/orgpolicy/v1/orgpolicy.proto#L290) of [Timestamp](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto) type. jsonpb expects Timestamp in ["{year}-{month}-{day}T{hour}:{min}:{sec}[.{frac_sec}]Z"](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto#L101) string format while CAI produces it in [the original format with nanos and seconds as integers](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto#L127).

On the other hand, while `json.Unmarshal` can handle org policies, it cannot handle access policies with `oneof` field.

Hence trying both approach while converting.

Reply:
Got it. I'd like to avoid putting critical code paths in an if statement though, instead could we refactor slightly to a separate function like this:

func unMarshallAsset() {
  err := jsonpb.Unmarshaler{AllowUnknownFields: true}
  if err == nil, return asset
  err = json.Unmarshal
  if err == nil, return asset
  return nil, err

Reply:
addressed in 1e1a678